File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/TopInterview2Round/mergeTwoLists_21.java
package algo.tzashinorpu.SpecialSubject.TopInterview2Round;

import algo.tzashinorpu.SpecialSubject.ListNode;

public class mergeTwoLists_21 {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummyNode = new ListNode();
        ListNode root = dummyNode;
        while (list1 != null && list2 != null) {
            if (list1.val < list2.val) {
                root.next = new ListNode(list1.val);
                list1 = list1.next;
            } else {
                root.next = new ListNode(list2.val);
                list2 = list2.next;
            }
            root = root.next;
        }
        if (list1 != null) root.next = list1;
        if (list2 != null) root.next = list2;
        return dummyNode.next;
    }
}

File: Clion/labuladong/chapter0/linklist/mergeTwoLists_21.cpp
//
// Created by TzashiNorpu on 5/31/2023.
//
#include "../../header/header.h"

using namespace labuladong;

ListNode *Solution::mergeTwoLists(ListNode *list1, ListNode *list2) {
  auto dummy = new ListNode(-2);
  auto x = new ListNode(-1);
  dummy->next = x;
  while (list1 && list2) {
    if (list1->val < list2->val) {
      x->next = list1;
      list1 = list1->next;
    } else {
      x->next = list2;
      list2 = list2->next;
    }
    x = x->next;
  }
  if (list2) { x->next = list2; }
  if (list1) { x->next = list1; }
  return dummy->next->next;
}